File: trustdevice/src/main/java/cn/tongdun/mobrisk/core/collectors/SensorsInfoCollector.java
package cn.tongdun.mobrisk.core.collectors;

import android.hardware.Sensor;
import android.hardware.SensorManager;

import org.json.JSONArray;
import org.json.JSONObject;

import java.util.List;

/**
 * @description: Sensors Info
 * @author: wuzuchang
 * @date: 2022/12/6
 */
public class SensorsInfoCollector {

    private SensorManager mSensorManager;

    public SensorsInfoCollector(SensorManager sensorManager) {
        mSensorManager = sensorManager;
    }

    public String getSensorList() {
        JSONArray sensorArray = new JSONArray();
        try {
            List<Sensor> sensorList = mSensorManager.getSensorList(Sensor.TYPE_ALL);
            for (int i = 0; i < sensorList.size(); i++) {
                Sensor sensor = sensorList.get(i);
                JSONObject sensorInfo = new JSONObject();
                sensorInfo.put("name", sensor.getName());
                sensorInfo.put("vendor", sensor.getVendor());
                sensorInfo.put("type", sensor.getType());
                sensorArray.put(i, sensorInfo);
            }
        } catch (Exception ignored) {
        }
        return sensorArray.toString();
    }
}

File: trustdevice/src/main/java/cn/tongdun/mobrisk/core/collectors/SensorsInfoCollector.kt
package cn.tongdun.mobrisk.core.collectors

import android.hardware.Sensor
import android.hardware.SensorManager
import cn.tongdun.mobrisk.core.tools.executeSafe
import org.json.JSONArray
import org.json.JSONObject

/**
 * @description:Sensors Info
 * @author: wuzuchang
 * @date: 2023/5/15
 */
interface SensorsInfoInterface {
    fun getSensorList(): String
}

class SensorsInfoCollector(private val sensorManager: SensorManager) : SensorsInfoInterface {
    override fun getSensorList(): String {
        val sensorArray = JSONArray()
        executeSafe {
            val sensorList: List<Sensor> = sensorManager.getSensorList(Sensor.TYPE_ALL)
            for (i in sensorList.indices) {
                val sensor = sensorList[i]
                val sensorInfo = JSONObject()
                sensorInfo.put("name", sensor.name)
                sensorInfo.put("vendor", sensor.vendor)
                sensorInfo.put("type", sensor.type)
                sensorArray.put(i, sensorInfo)
            }
        }
        return sensorArray.toString()
    }
}